sys/posix/pthread/pthread_cond.c

```diff
@@ -122,7 +122,7 @@ int pthread_cond_wait(pthread_cond_t *cond, mutex_t *mutex)
 
 int pthread_cond_timedwait(pthread_cond_t *cond, mutex_t *mutex, const struct timespec *abstime)
 {
-    if ((cond == NULL) || (abstime->tv_sec < 0)) {
+    if (cond == NULL) {
```

Comment:
For some platforms `tv_sec` can be an int: https://github.com/RIOT-OS/RIOT/blob/master/cpu/atmega_common/include/sys/types.h#L38

So this is not equivalent to before.

Reply:
I know, but on some its `unsigned` - and that's the whole purpose of the struct: to hide actual implementation. See also [here](https://stackoverflow.com/questions/471248/what-is-ultimately-a-time-t-typedef-to)

Reply:
For platform with signed `tv_sec`, the evaluation of 'then` would be wrong with negative values.
So maybe it should work only with `struct timespec` for comparing.

Reply:
This file is also missing now.
I planned to ping the person who added this `abstime->tv_sec < 0` to know if it made sense to remove it, but as it was you, I think you know what you are doing :)